sdk/clientcore/core/README.md

```diff
@@ -67,53 +31,124 @@ add the direct dependency to your project as follows.
 <dependency>
     <groupId>io.clientcore</groupId>
     <artifactId>core</artifactId>
-    <version>1.0.0</version>
+    <version>1.0.0-beta.1</version>
 </dependency>
 ```
 [//]: # ({x-version-update-end})
 
 ## Key concepts
 
-The key concepts of Azure Core (and therefore all Azure client libraries using Azure Core) include:
+The key concepts of Core (and therefore all client libraries using Core) include:
 
 - Configuring service clients, e.g. configuring retries, logging, etc. (`HttpTrait<T>`, `ConfigurationTrait<T>`, etc.)
 - Accessing HTTP response details (`Response<T>`).
-- Calling long-running operations (`PollerFlux<T>`).
-- Paging and asynchronous streams (`ContinuablePagedFlux<T>`).
 - Exceptions for reporting errors from service requests consistently.
 - Abstractions for representing SDK credentials.
 - Operation timeouts
+- JSON and XML serialization
 
 These will be introduced by way of the examples presented below.
 
-## Examples
+### JsonSerializable
 
-### Accessing HTTP Response Details Using `Response<T>`
+`JsonSerializable` is used to define how an object is JSON serialized and deserialized using stream-style serialization
+where the object itself manages the logic for how it's handled. The interface provides an instance-based `toJson` API
+that handles writing the object to a `JsonWriter` and a static `fromJson` API which implementations must provide to
+define how to create an object by reading from a `JsonReader`, if an implementation isn't provided
+`UnsupportedOperationException` will be thrown.
 
-_Service clients_ have methods that call Azure services, we refer call these methods _service methods_.
+### JsonToken
 
-_Service methods_ can return a shared Azure Core type `Response<T>`. This type provides access to both the
-deserialized result of the service call and to the details of the HTTP response returned from the server.
+`JsonToken` is a basic enum that indicates the current state in a JSON stream.
 
-### HTTP pipelines with `HttpPipeline`
+### JsonReader
```

Comment:
Should we move Json and XML documentation to their own .md files and link them here? It keeps this README focused and not distract readers with a lot of serde docs.

Reply:
For now, I'll just leave it in the main MD file but if we find it noisy later on it can be moved.